modules/internal/ChapelLocale.chpl

```diff
@@ -247,6 +247,35 @@ module ChapelLocale {
       return "";
     }
 
+    //
+    // Support for different types of memory:
+    // large, random-access (low latency), and streaming (high bandwidth)
+    // Inspired by madvise()'s support for RANDOM and SEQUENTIAL access
+    //
+    pragma "no doc"
+    proc defaultMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc largeMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc randomMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
+    pragma "no doc"
+    proc streamingMemory() : locale {
+      _throwPVFCError();
+      return this;
+    }
+
```

Comment:
Just a follow-on comment re: what @mppf just said: I'm not that fond of the bitmask idea, because I don't see how it can express an ordering for the desired characteristics and because it seems more like a C-style solution to me than a Chapel-style one.  Isn't there a more Chapelrific way to do this?